app/src/main/java/com/huji/foodtricks/foodtricks/ImageAdapter.java

```diff
@@ -84,25 +118,32 @@ public long getItemId(int position) {
     public void setIsPressed(int id) {
         isPressed[id] = !isPressed[id];
     }
+
     public boolean getIsPressed(int id) {
         return isPressed[id];
     }
     // create a new ImageView for each item referenced by the Adapter
 
     public View getView(int position, View convertView, ViewGroup parent) {
-        ImageView imageView;
+        View customView;
+        LayoutInflater inflater = (LayoutInflater) mContext.
+                getSystemService(Context.LAYOUT_INFLATER_SERVICE);
+
         if (convertView == null) {
```

Comment:
i know it's like this in the guide, but maybe it'll be better to switch the cases:
`if (convertView != null) {
return convertView 
} ... 
(else logic)`